Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/deletionwarning/SubmissionDeletionWarningViewModel.kt

```diff
@@ -36,7 +37,8 @@ class SubmissionDeletionWarningViewModel @AssistedInject constructor(
     val registrationError = SingleLiveEvent<CwaWebException>()
 
     fun deleteExistingAndRegisterNewTest() = launch {
-        coronaTestRepository.removeTest(coronaTest.identifier)
+        val currentTest = submissionRepository.testForType(coronaTest.type).first()
+        coronaTestRepository.removeTest(currentTest!!.identifier)
```

Comment:
should you catch the error here ? so, this would not crash for the user

Reply:
I think it is not needed in this case since I only navigate to the warning deletion screen if there is another test of the same type stored.